packages/vda-web3-client/src/VeridaContractBase.ts

```diff
@@ -206,40 +209,45 @@ export class VeridaContract {
                 if (methodType === 'view') {
                     ret = await contract.callStatic[methodName](...params)
                 } else {
-                    if (this.logPerformance) {
-                        console.log('Node endpoint: ' + (<JsonRpcProvider> contract.provider).connection.url)
-                    }
+                    let gasConfig : Record<string, any> | undefined = {}
+                    // console.log("Params : ", params)
+
+                    const paramLen = params.length
+                    if (params !== undefined
```

Comment:
This is very difficult to read.

Perhaps try doing something like:

const gasConfig = _.merge(params, this.methodDefaults[methodName], this.webConfig.gasConfig)